src/app/functions/server/onerep.ts

```diff
@@ -0,0 +1,283 @@
+  first_name: string;
+  last_name: string;
+  city: string;
+  state: StateAbbr;
+  birth_date?: ISO8601DateString;
+  age?: ISO8601DateString;
+  phone_number?: E164PhoneNumberString;
+};
+export type CreateScanResponse = {
+  id: number;
+  profile_id: number;
+  status: "in_progress";
+  reason: "manual";
+  created_at: ISO8601DateString;
+  updated_at: ISO8601DateString;
+};
+export type OneRepMeta = {
+  current_page: number;
+  from: number;
+  last_page: number;
+  per_page: number;
+  to: number;
+  total: number;
+};
+export type Scan = {
+  id: number;
+  profile_id: number;
+  status: "in_progress" | "finished";
+  reason: "initial" | "monitoring" | "manual";
+};
+export type ListScansResponse = {
+  meta: OneRepMeta;
+  data: Scan[];
+};
+export type ScanResult = {
+  id: number;
+  url: string;
+  link: string;
+  profile_id: number;
+  first_name: string;
+  last_name: string;
+  middle_name: string;
+  age: string;
+  addresses: Array<{
+    city: string;
+    state: StateAbbr;
+    street?: string;
+    zip?: string;
+  }>;
+  relatives: string[];
+  phones: string[];
+  emails: string[];
+  data_broker: string;
+  status: RemovalStatus;
+  data_broker_id: number;
+  created_at: ISO8601DateString;
+  updated_at: ISO8601DateString;
+};
+export type ListScanResultsResponse = {
+  meta: OneRepMeta;
+  data: ScanResult[];
+};
+export type RemovalStatus =
+  | "new"
+  | "optout_in_progress"
+  | "waiting_for_verification"
+  | "removed";
+
+async function onerepFetch(
+  path: string,
+  options: Parameters<typeof fetch>[1] = {}
+) {
+  const url = "https://api.onerep.com" + path;
+  const headers = new Headers(options.headers);
+  headers.set(
+    "Authorization",
+    `Basic ${Buffer.from(`${AppConstants.ONEREP_API_KEY}:`).toString("base64")}`
+  );
+  headers.set("Accept", "application/json");
+  headers.set("Content-Type", "application/json");
+  return fetch(url, { ...options, headers });
+}
+
+export async function createProfile(profileData: ProfileData): Promise<number> {
+  const requestBody = {
+    first_name: profileData.first_name,
+    last_name: profileData.last_name,
+    birth_date: profileData.birth_date,
+    addresses: [
+      {
+        state: profileData.state,
+        city: profileData.city,
+      },
+    ],
+  };
+  const response = await onerepFetch("/profiles", {
+    method: "POST",
+    body: JSON.stringify(requestBody),
+  });
+  if (!response.ok) {
+    log.info(
+      `Failed to create OneRep profile: [${response.status}] [${response.statusText}]`
+    );
+    throw new Error(
+      `Failed to create OneRep profile: [${response.status}] [${response.statusText}]`
+    );
+  }
+
+  const savedProfile: {
+    id: number;
+    status: "active" | "inactive";
+    created_at: ISO8601DateString;
+    updated_at: ISO8601DateString;
+    url: string;
+  } = await response.json();
+  return savedProfile.id;
+}
+
+export function parseExposureScanData(body: ScanResult): ProfileData | null {
```

Comment:
It took me a little bit to understand what this was intended for (I had to review the older controller code)... I think in the latest code there isn't really a call for parsing the scan result and outputting `ProfileData`, we're treating them separately like this:

1. `ProfileData` is the data that users enter in the form (first/last name, city, state, dob)
2. `ScanResults` from the remote API call are stored as-is in our database

Profiles and scan results are stored in separate tables and can be joined by profile ID. There may be more than one scan per user (they are differentiated by ID), but I think we want to keep these as separate classes of data.

Reply:
That said, for the moment we are allowing `ProfileData` to contain an optional `age` or `birth_date` field to keep the existing code from breaking, but it's something we can likely clean up. I'm not sure if we will need this parsing code at that point, but it could be re-purposed as a runtime guard perhaps.